### ISSUE TITLE
Sequential/Diverging defaults

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/plotly.js#2925]
`colorway` is going to be a very useful parameter for the new Themes feature of Chart Studio, because by setting just that attribute and saving it in a template, one will be able to provide custom defaults for colors of new traces, plus pie charts.

I think it will be necessary to add the equivalent custom defaults for sequential and diverging colorscales (overriding the current built-in default white-to-red and blue-to-white-to-red that appear when you first try to color by data in the editor), such that a template can specify just these three to prepopulate the default colors in all three modes: categorical, sequential and diverging.

**Comment by nicolaskruchten** [comment]
To clarify: this would be two new keys in `layout`, something like `layout.sequentialscale` and `layout.divergingscale`which would be arrays of color-strings, and whose defaults would be the current defaults plotly.js uses for those kinds of scales when `colorscale` is not specified.

**Comment by nicolaskruchten** [comment]
Even better :)

**Comment by nicolaskruchten** [comment]
This now seems fairly necessary, as Themes support in Chart Studio Cloud doesn't seem to grab the Theme-provided sequential colorscales ... or at least plotly.js isn't offering them up as defaults in `fullData` ...? http://plot.ly/create and try to make a scatter plot colored by a column with a theme.

**Comment by jonmmease** [comment]
This would be great. Right now themes have to individually set defaults for every colorscale across all trace types.  It would be nice to be able to specify this in one place (or one place for sequential and one place for diverging).

**Comment by nicolaskruchten** [comment]
Could we assume that diverging is the concatenation of increasing and
decreasing sequential or do we need all three?

**Comment by jonmmease** [comment]
I'm not used to thinking of increasing and decreasing as separate colorscale types.  Could you elaborate?  I was picturing that the defaults would be for sequential and diverging colorscales.

**Comment by nicolaskruchten** [comment]
I’m with you Jon but see comment above from Alex :)

**Comment by jonmmease** [comment]
Ah, missed that.  @alexj how does plotly.js decide which scale to use?

**Comment by alexcjohnson** [comment]
Right, the negative sequential is unusual, we only use it when you have uniformly negative data and you ask for `autocolorscale` (If you don't explicitly ask for `autocolorscale` we default to the diverging blue->gray->red). The idea being that the goal with a sequential colorscale is generally to vary from a "weak" color representing small values to a "strong" one representing large values, and "larger" more often means "bigger absolute value" than ">"

We could simply flip the positive sequential scale if all the data are negative, but the original thought was that this would be misleading to users used to seeing that scale the other way, so it would be better to use a separate scale.

> Could we assume that diverging is the concatenation of increasing and decreasing sequential or do we need all three?

You're right that our defaults are *close* to that - negative sequential being blue (most negative) to gray (least negative) and positive being gray (least positive) to red (most positive), vs the diverging blue->gray->red. But if we *actually* used the two concatenated together you'd see that the sequential ones end at a brighter gray than the diverging one, so there's a clear cusp at the boundary - which the diverging one smooths out in the name of perceptual uniformity.
![concat vs rdbu](https://user-images.githubusercontent.com/2678795/47955105-4828e580-df69-11e8-83b2-8ebc31160b36.png)

Perhaps if you only specify diverging and sequential, the negative sequential scale defaults to the reversed positive sequential?

**Comment by nicolaskruchten** [comment]
If plotly.js uses all three il happy with being able to specify all three
and let the template authors decide the details later on. If a layout-wide
setting is easy it would be great to get this into Chart Studio before the
end of the month :)

**Comment by antoinerg** [comment]
I am about to submit a PR for this one. What would be good names for those 3 new attributes? 

cc @plotly/plotly_js

**Comment by nicolaskruchten** [comment]
`layout.colorscales.positivesequential`, `layout.colorscales.negativesequential`, `layout.colorscales.diverging` ?

**Comment by etpinard** [comment]
- `layout.colorscale.sequential`
- `layout.colorscale.sequentialminus` (similar to error bar's [`arrayminus`](https://codepen.io/etpinard/pen/XybjZG?editors=1010))
- `layout.colorscale.diverging`

**Comment by archmoj** [comment]
I am curious to know if one could apply diverging reversed?
When having e.g. a topography model it would be nice to have the option of presenting the mountains in red and the elevations below zero in blue.

**Comment by nicolaskruchten** [comment]
@archmoj these are just for defaults... you can definitely reverse the colorscale, there's a different parameter for that :)
